### DISCUSSION TITLE
Ability for recipes to install or base off of other recipes

### DISCUSSION THREAD

**Discussion by jamiedavenport** [blitz-js/blitz#1273] [Ideas]
For example, I have a recipe for [TailwindUI](https://github.com/jamiedavenport/blitzjs-tailwindui-recipe) and this installs TailwindCSS as part of the recipe. The installation is mostly the same as the official TailwindCSS recipe with some additional files and transformations. I want to discuss whether or not there's a better way to handle this situation, below are some of my thoughts:

## Duplicate Setup

This is the approach I've taken so far. Not very good if changes to the prerequisite recipe are required but if the two recipes might diverge then I suppose there's no harm in this.

## README instructions

Alternatively, we could assume in the recipe that the prerequisites have already been installed. This would be documented in the recipe README and require the dev to do slightly more work

```bash
blitz install tailwind
blitz install tailwindui

# OR

blitz install tailwind tailwindui
```

## Install recipe step

We could build this into the installer with an `addRecipeStep`.

```ts
addRecipeStep(recipe: string): IRecipeBuilder
```

**Comment by flybayer** [comment]
@jamiedavenport yes, we should definitely add support for recipes to base off other recipes!

**Comment by Dohxis** [comment]
I would love to have this feature too.

**Comment by jamiedavenport** [comment]
With any of the approaches above other than duplicating the recipe, I'm thinking we need a way to version control deps? Is that already supported @flybayer?

**Reply by flybayer** [reply]
No, but we can add that
